src/js/components/ParseSection.jsx

```diff
@@ -121,42 +128,106 @@ const ParseSection = ( {setLogs, hardwareType} ) => {
         }
 
         if ( !parseError ) {
-            try {
-                if ( commandType === COMMAND_TYPE_MTX ) {
-                    data = joobyCodec[commandType].message.fromFrame(
-                        joobyCodec.utils.getBytesFromHex(hex),
-                        {
-                            aesKey: joobyCodec.utils.getBytesFromHex(parameters.accessKey)
-                        }
-                    );
-                } else {
-                    data = joobyCodec[commandType].message.fromHex(
-                        hex,
-                        {
-                            hardwareType: getHardwareType(hardwareType),
-                            direction: commandType === COMMAND_TYPE_ANALOG ? Number(parameters.direction) : undefined
+            const bytes = joobyCodec.utils.getBytesFromHex(hex);
+            const codec = joobyCodec[commandType];
+
+            switch ( commandType ) {
+                case COMMAND_TYPE_MTX: {
+                    const aesKey = joobyCodec.utils.getBytesFromHex(parameters.accessKey);
+
+                    try {
+                        direction = directions.DOWNLINK;
+                        data = codec.message[directionNames[direction]].fromFrame(bytes, {aesKey});
+                    } catch ( error ) {
+                        parseError = error;
+                    }
+
+                    if ( parseError || data.type !== frameTypes.DATA_REQUEST ) {
+                        parseError = null;
+                        data = null;
+
+                        try {
+                            direction = directions.UPLINK;
+                            data = codec.message[directionNames[direction]].fromFrame(bytes, {aesKey});
+                        } catch ( error ) {
+                            parseError = error;
                         }
-                    );
+                    }
+
+                    break;
                 }
-            } catch ( error ) {
-                parseError = error;
+
+                case COMMAND_TYPE_ANALOG:
+                    try {
+                        direction = Number(parameters.direction);
+                        data = codec.message[directionNames[direction]].fromBytes(
+                            bytes,
+                            {
+                                hardwareType: getHardwareType(hardwareType)
```

Comment:
```suggestion
                                {hardwareType: getHardwareType(hardwareType)}
```

such simple (and rather short) cases without nesting I propose to write in a single line

Reply:
fixed